### PR TITLE
Move sip_ua and hide internals

### PR DESCRIPTION
**Pull request by rlsutton1**
I've divided this PR into 2 commits. Although neither commit is viable without the other, it helps to separate the simple and complex parts of this PR.

The first commit https://github.com/cloudwebrtc/dart-sip-ua/commit/acdba147f83f6f5a610d90af4630f2b79a879e9f simply changes paths and imports

The second commit https://github.com/cloudwebrtc/dart-sip-ua/commit/e0c474cf9f88f4cfa2e822ac62d151ae39374a81 is concerned with changing the listener mechanism between example and sip_ua_helper

### CODE REVIEW

File: lib/src/sip_ua_helper.dart [hunk 1]
```diff
@@ -146,55 +162,61 @@ class SIPUAHelper extends EventManager {
...
-      _handleCallState('unmuted', null, null, null, e.audio, e.video);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.UNMUTED, audio: e.audio, video: e.video));
     });
     eventHandlers.on(EventStream(), (EventStream e) async {
       // Wating for callscreen ready.
       Timer(Duration(milliseconds: 100), () {
-        _handleCallState('stream', null, e.stream, e.originator, null, null);
+        _notifyCallStateListeners(
+            CallState(CallStateEnum.STREAM, stream: e.stream));
```
Comment (cloudwebrtc) [inline lib/src/sip_ua_helper.dart]: Here, we need to pass the  `originator` parameter, you can know that `stream` is `local` or `remote` for proper display.

File: lib/src/sip_ua_helper.dart [hunk 2]
```diff
@@ -78,40 +101,42 @@ class SIPUAHelper extends EventManager {
...
+        _registerState = RegistrationStateEnum.UNREGISTERED;
         _registered = false;
-        _handleRegisterState('unregistered', event.response);
+        _notifyRegsistrationStateListeners(RegistrationStateEnum.UNREGISTERED);
       });
 
       this._ua.on(EventRegistrationFailed(), (EventRegistrationFailed event) {
         logger.debug('registrationFailed => ' + (event.cause));
-        _registerState = 'registrationFailed[${event.cause}]';
+        _registerState = RegistrationStateEnum
```
Comment (cloudwebrtc) [inline lib/src/sip_ua_helper.dart]: Need a way to send `code` and `cause` to the UI, telling why the registration failed.

File: lib/src/sip_ua_helper.dart [hunk 3]
```diff
@@ -146,55 +162,61 @@ class SIPUAHelper extends EventManager {
...
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.PROGRESS, originator: event.originator));
     });
 
     eventHandlers.on(EventFailed(), (EventFailed event) {
       logger.debug('call failed with cause: ' + (event.cause));
-      _handleCallState(
-          'failed', event.response, null, event.originator, null, null);
+      _notifyCallStateListeners(
+          CallState(CallStateEnum.FAILED, originator: event.originator));
```
Comment (cloudwebrtc) [inline lib/src/sip_ua_helper.dart]: Same as registration, Need a way to send `code` and `cause` to the UI, telling why the call failed.

Comment (cloudwebrtc) [inline lib/src/sip_ua_helper.dart]: Same as registration, Need a way to send `code` and `cause` to the UI, telling why the call failed.

Comment (rlsutton1) [reply lib/src/sip_ua_helper.dart]: I'm adding cause, but it appears that code is not passed and not available in the EventFailed

File: lib/src/sip_ua_helper.dart (hunk 1 above)
Comment (rlsutton1) [reply lib/src/sip_ua_helper.dart]: adding originator

File: lib/src/sip_ua_helper.dart (hunk 2 above)
Comment (rlsutton1) [reply lib/src/sip_ua_helper.dart]: code is not available, and as only cause is available I'll just add an extra parameter to the listener like this...

 void registrationStateChanged(RegistrationStateEnum state, String cause);

**Comment by rlsutton1** [comment]
I've made changes and updated the PR

**Comment by cloudwebrtc** [comment]
LGTM

**Comment by rlsutton1** [comment]
thanks

**Pull request merged**